File: backend/src/main/java/tn/tawiniya/tounisiya/entity/TrainingCenter.java
package tn.tawiniya.tounisiya.entity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
@Entity
@Table(name = "training_centers")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TrainingCenter {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;
    @Column(nullable = false)
    private String name;
    @Column(columnDefinition = "TEXT")
    private String description;
    private String address;
    private String phone;
    private String email;
    private String website;
    // Chemin public du logo, ex: /uploads/training-centers/xxx.jpg
    private String logoPath;
    private String openingHours;
    @OneToMany(mappedBy = "trainingCenter", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    @Builder.Default
    private List<TrainingCourse> courses = new ArrayList<>();
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
    @PrePersist
    protected void onCreate() {
        LocalDateTime now = LocalDateTime.now();
        this.createdAt = now;
        this.updatedAt = now;
    }
    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/types/trainingCenter.ts
export interface TrainingCourse {
  id: number;
  title: string;
  description: string | null;
}
export interface TrainingCourseInput {
  title: string;
  description?: string;
}
export interface TrainingCenter {
  id: number;
  ownerId: number;
  name: string;
  description: string | null;
  address: string | null;
  phone: string | null;
  email: string | null;
  website: string | null;
  logoPath: string | null;
  openingHours: string | null;
  courses: TrainingCourse[];
  createdAt: string;
  updatedAt: string;
}
export interface TrainingCenterInput {
  name: string;
  description?: string;
  address?: string;
  phone?: string;
  email?: string;
  website?: string;
  openingHours?: string;
  courses?: TrainingCourseInput[];
}